test/spirv_param_decorations.ll

```diff
@@ -0,0 +1,50 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir"
+
+; Function Attrs: convergent nounwind
+define spir_kernel void @k(float %a, float %b, float %c) #0 !kernel_arg_addr_space !4 !kernel_arg_access_qual !5 !kernel_arg_type !6 !kernel_arg_type_qual !7 !kernel_arg_base_type !6 !spirv.ParameterDecorations !14 {
```

Comment:
Could you please a test case, when we have both decoration metadata attached to the kernel and restrict/volatile specifiers on the kernel arguments to check if the translation can handle interfering representations. This comment is applicable to another test.

Reply:
I have added a test test/spirv_param_decorations.ll that tests translation where `!kernel_arg_type_quals` specifies an argument to be volatile and `!spirv.ParameterDecorations` specifies noalias on the same argument. Note that `restrict` in `!kernel_arg_type_quals` infers noalias and not Restrict decoration. The test shows that the qualifiers are merged, i.e. the resulting `!kernel_arg_type_quals` has `!"volatile restrict"` and the re-translated decorator metadata list has both noalias and volatile decorator nodes. Is this in line with what you were thinking?

Reply:
Yes, thanks!